swarm_notif_test.go

```diff
@@ -16,18 +16,32 @@ import (
 func TestNotifications(t *testing.T) {
 	const swarmSize = 5
 
+	notifiees := make([]*netNotifiee, swarmSize)
+
 	ctx := context.Background()
 	swarms := makeSwarms(ctx, t, swarmSize)
 	defer func() {
-		for _, s := range swarms {
-			s.Close()
+		for i, s := range swarms {
+			select {
+			case <-notifiees[i].listenClose:
+				t.Error("should not have been closed")
+			default:
+			}
+			err := s.Close()
+			if err != nil {
+				t.Error(err)
+			}
+			select {
```

Comment:
Should _probably_ fail the test on a timed context rather than the default case as the `ListenClose ` notif is sent inside a go-routine that `Swarm.Close` does not synchronise with.

Reply:
I take my word back. `Swarm.Close` waits of all refs to complete, one of which is completed by the go-routine that sends the `ListenClose ` notification.

Apporved.